### DISCUSSION TITLE
getInputFiles() does not return with the fragment

### DISCUSSION THREAD

**Discussion by zahramesri** [GreenleafLab/ArchR#1400] [Questions / Documentation]
Hi,

The getInputFiles() does not return with the fragment file for further analysis. I have come across this https://github.com/GreenleafLab/ArchR/issues/87 and installed Fortran (gfortran) and XQuartz, but did not resolve the issue. Please let me know if you have any suggestions.

Here is the code:

> inputFiles <- list.files("inputFile")
> inputFiles
> [1] "fragments.tsv.gz"

> inputFiles <- getInputFiles("inputFile")
> inputFiles
> character(0)

Thank you,
Zahra

**Comment by rcorces** [comment]
This is not the intended use of this function.
![image](https://user-images.githubusercontent.com/10343046/165151028-73927603-6613-46c9-9ebe-51e38b11856b.png)

**Reply by zahramesri** [reply]
Thank you for your response. Could you elaborate please?

**Reply by rcorces** [reply]
The argument to `getInputFiles()` is a file path. This is clearly stated in the function documentation.
https://www.archrproject.com/reference/getInputFiles.html

**Reply by zahramesri** [reply]
Thank you for your reply.  I understand that the input argument should be the path in which appropriate files are stored. 

As you can see here: 
> path
[1] "/Users/zahra/Desktop/ArchR"
> inputFiles <- getInputFiles(paths = path)
> inputFiles
character(0)

I am giving the path as argument, however, when I call inputFiles the return in empty. I hope this clarifies the issue I am facing. Please let me know if I can provide further information.  

Thank you so much,

**Reply by zahramesri** [reply]
I resolved the issue.

**Reply by rustycandlewick** [reply]
Hi, can you please provide how you solved this (I am facing a similar problem). I am trying to get multiple fragment files from the path so should i make arrow files individually or is this the best way to do that?
